sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -461,51 +597,301 @@ is `name`.
-      syclex::source_language::/*some-language*/,
+      syclex::source_language::sycl,
       source);
 
+  // Compile the kernel.  There is no need to use the "registered_kernel_names"
+  // property because the kernel is declared extern "C".
   sycl::kernel_bundle<sycl::bundle_state::executable> kb_exe =
     syclex::build(kb_src);
 
-  // Get the kernel via its name.  The "kernel-name" is a stand-in for the
-  // actual kernel name in the source string.
-  sycl::kernel k = kb_exe.ext_oneapi_get_kernel("kernel-name");
+  // Get the kernel via its compiler-generated name.
+  sycl::kernel iota = kb_exe.ext_oneapi_get_kernel("iota");
 
+  float *ptr = sycl::malloc_shared<float>(NUM, q);
   q.submit([&](sycl::handler &cgh) {
-    // Any arguments for the kernel must be set manually.
-    cgh.set_args(/*...*/);
+    // Set the values of the kernel arguments.
+    cgh.set_args(3.14f, ptr);
 
-    // Launch the kernel according to its type.
-    // This assumes a simple "range" kernel.
-    cgh.parallel_for(sycl::range{1024}, k);
-  });
+    // Launch the kernel according to its type, in this case a simple
+    // "range" kernel.
+    cgh.parallel_for({NUM}, iota);
+  }).wait();
 }
 ```
+
+=== Disambiguating overloaded kernel functions
+
+This example demonstrates how to use the `registered_kernel_names` property to
+disambiguate a kernel function that has several overloads.
+
+```
+#include <sycl/sycl.hpp>
+namespace syclex = sycl::ext::oneapi::experimental;
+static constexpr size_t NUM = 1024;
+
+int main() {
+  sycl::queue q;
+
+  // The source code for two kernels defined as overloaded functions.
+  std::string source = R"""(
+    #include <sycl/sycl.hpp>
+    namespace syclex = sycl::ext::oneapi::experimental;
+
+    SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclex::range_kernel<1>))
+    void iota(float start, float *ptr) {
+      size_t id = syclex::this_kernel::get_id();
+      ptr[id] = start + static_cast<float>(id);
+    }
+
+    SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclex::range_kernel<1>))
+    void iota(int start, int *ptr) {
+      size_t id = syclex::this_kernel::get_id();
+      ptr[id] = start + static_cast<int>(id);
+    }
+  )""";
+
+  // Create a kernel bundle in "source" state.
+  sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source> kb_src =
+    syclex::create_kernel_bundle_from_source(
+      q.get_context(),
+      syclex::source_language::sycl,
+      source);
+
+  // Compile the kernel.  Because there are two overloads of "iota", we need to
+  // use a C++ cast to disambiguate between them.  Here, we are selecting the
+  // "int" overload.
+  std::string iota_name{"(void(*)(int, int*))iota"};
+  sycl::kernel_bundle<sycl::bundle_state::executable> kb_exe = syclex::build(kb_src,
+    syclex::properties{syclex::registered_kernel_names{iota_name}});
+
+  // Get the kernel by passing the same string we used to construct the
+  // "registered_kernel_names" property.
+  sycl::kernel iota = kb_exe.ext_oneapi_get_kernel(iota_name);
```

Comment:
I was thinking of something like 
`sycl::kernel iota = kb_exe.ext_oneapi_get_kernel( syclex::get_spirv_kernel_name("iota") );`

Where `get_spirv_kernel_name` would return the mangled kernel name that matches.   Of course, this doesn't handle different overloads of `iota`.  A simple solution like that might work, with a fallback to something like `iota_name"(void(*)(int*, int*))iota"};` when using overloads.  Otherwise, it seems like we are needlessly complicating it for everyone who doesn't use overloads.